src/App.tsx

```diff
@@ -1,14 +1,40 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [setTodo, setAll] = useState('all');
+  const [setTodoId, setTodoList] = useState<Todo | null>(null);
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  const getVisibleTodos = todos.filter(todo => {
+    const getVisible = todo.title.toLowerCase().includes(query.toLowerCase());
+
+    if (setTodo === 'active') {
+      return getVisible && todo.completed === false;
+    }
+
+    if (setTodo === 'completed') {
+      return getVisible && todo.completed === true;
+    }
+
+    return getVisible;
+  });
+
+  getTodos().then(tod => {
```

Comment:
looks like it causes an infinity cycle (as you see your app is stucking after any changes)
do it in `useEffect` hook just  after the first render